lib/mix/lib/mix.ex

```diff
@@ -428,15 +430,25 @@ defmodule Mix do
     Mix.State.put(:debug, debug)
   end
 
+  @doc """
+  Raises a Mix error that is nicely formatted, defaulting to exit code `1`.
+  """
+  @spec raise(binary) :: no_return
+  def raise(message) do
+    __MODULE__.raise(message, exit_code: 1)
+  end
+
   @doc """
   Raises a Mix error that is nicely formatted.
 
   ## Options
 
     * `:exit_code` - defines exit code value, defaults to `1`
+
   """
-  @spec raise(binary, exit_code: non_neg_integer()) :: no_return
-  def raise(message, opts \\ []) when is_binary(message) do
+  @doc since: "1.12.0"
+  @spec raise(binary, [option]) :: no_return when option: {:exit_code, non_neg_integer()}
```

Comment:
The keyword syntax from before is clearer. Thoughts?

Reply:
I remember @fishcakez saying something about how that makes it a breaking change to add options, but at this point I have no idea. @josevalim should I roll this back?

Reply:
That's for types but not for specs. :) In types, you can't augment it with new options if it is a keyword list.

Reply:
@josevalim got it. I reverted this line.